src/Raven.Server/Documents/Replication/Incoming/IncomingReplicationHandler.cs

```diff
@@ -685,7 +663,7 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
             protected virtual void SaveSourceEtag(DocumentsOperationContext context)
             {
                 context.LastReplicationEtagFrom ??= new Dictionary<string, long>();
-                if (_replicationType != ReplicationLatestEtagRequest.ReplicationType.Sharded)
+                if (_replicationInfo.SourceDatabaseId != null)
```

Comment:
hmm we use the `IncomingReplicationHandler` for both sharded and non-sharded replication?